tools/commit-message-lint

```diff
@@ -5,10 +5,12 @@
 
 # The rules can be found in /.gitlint
 
+repository="zulip/zulip"
+
 if [[ "
 $(git remote -v)
 " =~ '
-'([^[:space:]]*)[[:space:]]*(https://github\.com/|ssh://git@github\.com/|git@github\.com:)zulip/zulip(\.git|/)?\ \(fetch\)'
+'([^[:space:]]*)[[:space:]]*(https://github\.com/|ssh://git@github\.com/|git@github\.com:)$repository(\.git|/)?\ \(fetch\)'
```

Comment:
Same comment as in the python-zulip-api copy -- have you tested this? I 'm not sure whether the variable is being interpolated correctly, and I can't tell if you've tested that detail from the discussion.

@andersk FYI in case the quoting issue proves complicated.

Reply:
I am quite sure this works. Though I don't mind waiting for the review of @andersk.

Reply:
more over. I think it is better to write strings in double quotes as they are static where as the regex and stuffs like that can be in single quotes.

Reply:
Hi, sorry I read about it, and apparently, it shouldn't work, but probably worked while testing because of some reason, I am taking a look at fixing this.

Reply:
Sorry, but it actually works, Will wait for @andersk to know more about how it works!!

Reply:
Think, it is because of the single quotes enclosing the regex in single quotes.

Reply:
This is not in single quotes. Only the newlines themselves are in single quotes.

However, `$repository` should be double-quoted as `"$repository"` so that Bash interprets it as a string rather than a regex.

```suggestion
'([^[:space:]]*)[[:space:]]*(https://github\.com/|ssh://git@github\.com/|git@github\.com:)"$repository"(\.git|/)?\ \(fetch\)'
```

Reply:
Thanks for this, I also thought that this might be the reason.